examples/finetune_resnet/components.py

```diff
@@ -0,0 +1,81 @@
+import os
+from collections import defaultdict
+from pickletools import optimize
+from typing import Any, Dict, List, Tuple
+
+import torch
+import torch.nn as nn
+from torch.optim import Adam
+from torchvision import datasets, models, transforms
+
+from tango import Step
+from tango.integrations.torch import DataCollator, Model, Optimizer, model
+
+Optimizer.register("torch_adam")(Adam)
+
+
+def set_parameter_requires_grad(model, feature_extracting):
+    if feature_extracting:
+        for param in model.parameters():
+            param.requires_grad = False
+
+
+@Model.register("resnet_ft")
+class ResNetWrapper(Model):
+    def __init__(self, num_classes: int, feature_extract: bool, use_pretrained: bool):
+        super().__init__()
+        self.model_ft = models.resnet18(pretrained=use_pretrained)
+        set_parameter_requires_grad(self.model_ft, feature_extract)
+        num_ftrs = self.model_ft.fc.in_features
+        self.model_ft.fc = nn.Linear(num_ftrs, num_classes)
+        self.loss_fn = nn.CrossEntropyLoss()
+
+    def forward(self, image, label) -> Dict[str, torch.Tensor]:
```

Comment:
Let's add type hints and make `label` optional so that we can use this model in inference.

```suggestion
    def forward(self, image:  torch.Tensor, label: Optional[torch.Tensor] = None) -> Dict[str, torch.Tensor]:
```

You have to handle creating predictions with `argmax` below when `label` is `None`.